mantidproject/mantid
# TOFTOF GUI: add temperature fields

**Description of work.**
added: 
 - temperature input fields for Vanadium runs and empty-can runs (optional)
 - a temperature input column to the data-runs table (optional)

changed:
 - moved `DataTableModel` and `DataTableView` classes from `TOFTOFSetupWidget` class to their own file.
 - Script generator sets `temperature` to ca constant value if the temperature for a run is set, otherwise it calculates it from the run log data

**To test:**
Start the TOFTOF data reduction GUI: Interfaces->Direct->DGS Reduction. Choose facility MLZ and instrument TOFTOF.

Get the test data
[testdata.zip](https://github.com/mantidproject/mantid/files/1356265/testdata.zip)

and make sure that the sample-log for 'temperature' is added to the Vanadium runs workspace is set to a the value set in the `T (K)`field, if (and only if) the field is not empty. Otherwise it should get the temperature from the runs own LogData.
For empty-can runs (EC-Runs) and each row of data-runs it should behave accordingly.

Fixes #22588. 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
